ntbs-service/Pages/ContactDetails/Edit.cshtml.cs

```diff
@@ -58,7 +61,7 @@ public async Task<IActionResult> OnPostAsync()
             }
 
             await _userRepository.UpdateUserContactDetails(ContactDetails);
-            return RedirectToPage("/ContactDetails/Index");
+            return RedirectToPage("/ContactDetails/Index", new {username = Username});
```

Comment:
This is weird - why aren't we redirecting to `/ContactDetails/{Username}/Index`?

Reply:
I was getting a `No page named 'ContactDetails/Username/Index' matches the supplied values.` error. Something similar to this https://www.codeproject.com/Questions/5164350/How-do-I-redirect-from-a-page-in-one-folder-to-a-p

Reply:
Oh interesting, I think the path here is a file path rather than an application route - I'm a bit surprised by this, but this makes sense then